src/euphorie/client/browser/login.py

```diff
@@ -88,16 +88,44 @@ def login(self, account, remember):
                 120 * 24 * 60 * 60
             )  # noqa: E501
 
-    def transferGuestSession(self, account_id):
-        """Transfer session(s) from guest account to an existing user account"""
-        if not account_id:
+    def transferGuestSession(self):
+        """Transfer session(s) from guest account to an existing user account
+
+        The guest account is expected to go the login form as an authenticated user.
+
+        For this reason we can know from the session plugin who he is even
+        if during the request process we are now logged in as a new user.
+        """
+        plugin = self.context.acl_users.session
+        old_credentials = plugin.authenticateCredentials(
+            plugin.extractCredentials(self.request)
+        )
+        if old_credentials is None:
+            # We came here as anonymous user, no need to proceed
             return
-        account = get_current_account()
+
+        old_authenticated_account_id = old_credentials[0]
+
+        if (
+            Session.query(model.Account)
+            .filter(
+                model.Account.account_type == config.CONVERTED_ACCOUNT,
```

Comment:
I am not sure about this line, maybe it should be:
```suggestion
                model.Account.account_type == config.GUEST_ACCOUNT,
```
It might also be that the method should not even exist.